components/console/esp_console.h

```diff
@@ -156,6 +157,15 @@ esp_err_t esp_console_deinit(void);
  */
 typedef int (*esp_console_cmd_func_t)(int argc, char **argv);
 
+/**
+ * @brief Console command main function
+ * @param argc number of arguments
+ * @param argv array with argc entries, each pointing to a zero-terminated string argument
+ * @param context a user context given at invocation.
+ * @return console command return code, 0 indicates "success"
+ */
+typedef int (*esp_console_cmd_func_context_t)(int argc, char **argv, void *context);
```

Comment:
BTW: I prefer the context to be placed as 1st parameter, please tell me what you prefer.

Reply:
yes please, context always first